### PR TITLE
Add completion timeout configurability

### PR DESCRIPTION
**Pull request by disa6302**
*Issue #, if available:*

*Description of changes:*
- Add configurability for curl call connection and completion timeouts. 
- Fix compiler warning for sprintf in several unit tests
- Add unit test for version (client info)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/client/include/com/amazonaws/kinesis/video/client/Include.h [hunk 1]
```diff
@@ -1148,6 +1154,8 @@ typedef struct __ClientInfo {
     // Version of the struct
     UINT32 version;
 
+    // ------------------------------- V0 compat ----------------------
```
Comment (hassanctech) [inline src/client/include/com/amazonaws/kinesis/video/client/Include.h:1157]: The previous compat lines were correct.  They indicate that everything above the comment line is compatible with that version.  V0 included up until the log metric.  See this commit is where `ClientInfo` was introduced and in the same commit the `CLIENT_INFO_CURRENT_VERSION` was set to `0`.

**Review by MushMal — DISMISSED** [review]
Other than @hassanctech comment, I would recommend adding a couple of version-specific tests.

File: src/client/include/com/amazonaws/kinesis/video/client/Include.h (hunk 1 above)
Comment (disa6302) [reply src/client/include/com/amazonaws/kinesis/video/client/Include.h]: Fixed

**Comment by disa6302** [comment]
> Other than @hassanctech comment, I would recommend adding a couple of version-specific tests.

Done

File: src/client/include/com/amazonaws/kinesis/video/client/Include.h [hunk 2]
```diff
@@ -1484,6 +1495,9 @@ struct __ServiceCallContext {
 
     // Authentication info
     PAuthInfo pAuthInfo;
+
+    // -------- V1 compat --------
```
Comment (hassanctech) [inline src/client/include/com/amazonaws/kinesis/video/client/Include.h:1499]: This should be V0 compat, marking the end of V0, to match the format in our other versioned structs.  See for example `ClientInfo` and `StreamCaps`

**Review by hassanctech — APPROVED** [review]
LGTM!

**Review by jdelapla — CHANGES_REQUESTED** [review]
With a core struct being changed we also need the associated JNI change before we take this in.

File: src/client/src/InputValidator.c [hunk 3]
```diff
@@ -326,6 +333,23 @@ VOID fixupClientInfo(PClientInfo pClientInfo, PClientInfo pOrigClientInfo)
...
+        pClientInfo->serviceCallCompletionTimeout = SERVICE_CALL_DEFAULT_TIMEOUT;
+    }
+
+    if(pClientInfo->serviceCallConnectionTimeout == 0 || !IS_VALID_TIMESTAMP(pClientInfo->serviceCallConnectionTimeout)) {
+        // Although curl sets default if the value is 0, the default is 300 seconds (5 minutes) which is very high
+        DLOGW("Connection timeout should be non zero...setting to default");
+        pClientInfo->serviceCallConnectionTimeout = SERVICE_CALL_DEFAULT_CONNECTION_TIMEOUT;
+    }
+
+    if(pClientInfo->serviceCallCompletionTimeout == 0 || !IS_VALID_TIMESTAMP(pClientInfo->serviceCallCompletionTimeout)) {
```
[the remark this answers is not in the document]
Comment (jdelapla) [reply src/client/src/InputValidator.c:348]: Nevermind, my eyes glazed over the completion != connection

**Comment by hassanctech** [comment]
> With a core struct being changed we also need the associated JNI change before we take this in.

I don't think so the struct is versioned, so it retains backwards compat with older versions of the struct, if it encounters older versions (which it would if invoked from what is in JNI) then it would use the defaults.

**Comment by disa6302** [comment]
> > With a core struct being changed we also need the associated JNI change before we take this in.
> 
> I don't think so the struct is versioned, so it retains backwards compat with older versions of the struct, if it encounters older versions (which it would if invoked from what is in JNI) then it would use the defaults.

This is true. But I think @jdelapla was trying to achieve feature parity between JNI and Java.

**Review by jdelapla — APPROVED** [review]

**Pull request merged**